### ISSUE TITLE
find user incompatible with split / join nodes

### ISSUE THREAD

**Issue by PaulWieland** [NoeSamaille/node-red-contrib-activedirectory#12]
If the `find user` node isn't able to find the specified username, the node throws an error in the debug console and doesn't return any message to the next node in the flow. 

Because the `join` node waits for all parts to be received, it never completes.

The node should be updated to always return a message, even if the user was not found. Silencing errors should also be an option (or don't throw errors, and instead put an error message in `msg.ad_error`)

**Comment by Xenokrates** [comment]
Hi @PaulWieland,

with the last commits in this repo the `find user` node throughs an error with the message.
So at a work around it would be possible to catch the error with a `catch` node, clean up the payload and reinject the message, so that the `join`node can complete.

Nevertheless this would be only a work around.
I would propose to change the default behaviour of the nodes to just send an empty payload, if the query gets an empty result.
An AD/LDAP is only a database and an empty resultset to a query is - as a would suggest -not an error.

But this would change the designed behaviour of all the nodes.
@NoeSamaille - Would you agree on the functional change of the nodes to reply with an empty payload and not with an error, if  no user/group/... was found ?

I would implement these small changes.

Best regards

**Comment by Xenokrates** [comment]
Hi @PaulWieland,

I just run into another issue from the underlying node `activdirectory2`, that is still not solved:
If your credentials are wrong, the callback fires twice and messes up your message flow. ([See here.](https://github.com/jsumners/node-activedirectory/issues/46#issuecomment-884126377))

But I have another question:
What shall be the `msg.payload`, when no user is found ?
Wenn I set it to `null` I get an error in the debug windows with the debug node.
May be setting it to an empty string is ok. Or set it to an empty option ?
Some suggestions ?

Regards

**Comment by PaulWieland** [comment]
@Xenokrates Actually - I think `msg.payload` can be undefined. The only thing that must happen is that the `msg` must be returned. I don't have time to test this morning, but you should be able to test an undefined payload by deleting the `payload` property from an inject node and then fire it. Use a debug node and display the entire `msg` object to see the content.

**Comment by Xenokrates** [comment]
Actually - I think msg.payload can be undefined.
No, I tried this with `msg.payload = null;`and it gives an awful TypeError on the debug panel thrown from the `debug`-node.
<img width="427" alt="Screenshot 2021-07-21 155217" src="https://user-images.githubusercontent.com/16815342/126500158-6b25cbb3-27b8-484a-b04e-d70827a3107f.png">

I think, `msg.payload = new Object();` would be a good solution.
It is easy to analyse with a 'is empty' value rule in a switch node.

**Comment by PaulWieland** [comment]
@Xenokrates `null` is not the same as `undefined` or not set. I think you set the entire `msg` object to null, which is not okay.

Try this flow - it proves that payload can be `null` or `undefined` and it works just fine.

```
[
    {
        "id": "d06208a8.222788",
        "type": "inject",
        "z": "fbf80cd6.d612d8",
        "name": "",
        "props": [
            {
                "p": "payload"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "null",
        "payloadType": "json",
        "x": 310,
        "y": 100,
        "wires": [
            [
                "e5914ecc.172478"
            ]
        ]
    },
    {
        "id": "c028ee3e.7a2808",
        "type": "inject",
        "z": "fbf80cd6.d612d8",
        "name": "undefined",
        "props": [],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "x": 300,
        "y": 140,
        "wires": [
            [
                "e5914ecc.172478"
            ]
        ]
    },
    {
        "id": "e5914ecc.172478",
        "type": "debug",
        "z": "fbf80cd6.d612d8",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "true",
        "targetType": "full",
        "statusVal": "",
        "statusType": "auto",
        "x": 450,
        "y": 140,
        "wires": []
    },
    {
        "id": "b4c2e88d.82618",
        "type": "inject",
        "z": "fbf80cd6.d612d8",
        "name": "string",
        "props": [
            {
                "p": "payload"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "test",
        "payloadType": "str",
        "x": 310,
        "y": 180,
        "wires": [
            [
                "e5914ecc.172478"
            ]
        ]
    }
]
```

**Comment by Xenokrates** [comment]
@PaulWieland 
I have tried your example flows. 
They lead to the desired outputs, but I struggled to do this in JavaScipt code.

I do not know which is the right way to set the `msg.payload` to `null`.

All the following fails:
````
msg.payload =null;
msg.payload = JSON.parse(null);
````
I could not find out, how the `inject`node is smuggling this value into the `payload`attribute.

So the easiest approach seems to be to delete the payload:
````
delete msg.payload;
````
